content/docs/rfcs/36/README.md

```diff
@@ -0,0 +1,642 @@
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field contains an array of multiaddresses.
+The multiaddresses are `string`s.
+
+For example:
+
+```json
+{
+  "result": [
+    "/ip4/127.0.0.1/tcp/30303",
+    "/ip4/1.2.3.4/tcp/30303",
+    "/dns4/waku.node.example/tcp/8000/wss"
+  ]
+}
+```
+
+## Connecting to peers
+
+### `extern char* waku_add_peer(int nodeId, char* address, char* protocolId)`
+
+Add a node multiaddress and protocol to the waku node's peerstore.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+2. `char* address`: A multiaddress to reach the peer being added.
+3. `char* protocolId`: A protocol we expect the peer to support.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field contains the peer ID as a base58 `string` of the peer that was added.
+
+For example:
+
+```json
+{
+  "result": "QmWjHKUrXDHPCwoWXpUZ77E8o6UbAoTTZwf1AD1tDC4KNP"
+}
+```
+
+### `extern char* waku_dial_peer(int nodeId, char* address, int timeoutMs)`
+
+Dial peer using a multiaddress.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+2. `char* address`: A multiaddress to reach the peer being dialed.
+3. `int timeoutMs`: Timeout value in milliseconds to execute the call.
+   If the function execution takes longer than this value,
+   the execution will be canceled and an error returned.
+   Use `0` for no timeout.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field is set to `null`.
+
+For example:
+
+```json
+{
+   "result": null
+}
+```
+
+### `extern char* waku_dial_peerid(int nodeId, char* peerId, int timeoutMs)`
+
+Dial peer using its peer ID.
+
+**Parameters**
+
+1. `int nodeId`: The node identifier obtained from a successful execution of [`waku_new`](#extern-char-waku_newchar-configjson).
+2. `char* peerID`: Peer ID to dial.
+   The peer must be already known.
+   It must have been added before with [`waku_add_peer`](#extern-char-waku_add_peerint-nodeid-char-address-char-protocolid)
+   or previously dialed with [`waku_dial_peer`](#extern-char-waku_dial_peerint-nodeid-char-address-int-timeoutms).
+3. `int timeoutMs`: Timeout value in milliseconds to execute the call.
+   If the function execution takes longer than this value,
+   the execution will be canceled and an error returned.
+   Use `0` for no timeout.
+
+**Returns**
+
+A [`JsonResponse`](#jsonresponse).
+If the execution is successful, the `result` field is set to `null`.
+
+For example:
+
+```json
+{
+   "result": null
+}
+```
+
+### `extern char* waku_close_peer(int nodeId, char* address)`
```

Comment:
Is this API needed? I assume working with the peer id should be enough.

Reply:
yeah. I think I'll even remove that from go-waku as well.
What do you think of renaming `waku_close_peerId` to `waku_close_peer`?